google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/Acl.java

```diff
@@ -119,6 +122,11 @@ public Type getType() {
     abstract Access toPb();
 
     static Entity fromPb(Access access) {
+      if (access.getDataset() != null) {
+        return new Dataset(
```

Comment:
minor constructor surprise for me, until I realized we're dealing with an ACL.Dataset and not a Dataset.  I suspect this is just local ambiguity; any other places where we need to consider the naming?

Reply:
Yeah I agree it's a little confusing so I renamed it to be more explicit: `DatasetAclEntity`.